### PR TITLE
Make CommunityModules a test-time dependency of TLC.

### PR DESCRIPTION
**Pull request by lemmy**
This is https://github.com/tlaplus/CommunityModules/releases/tag/202412190202 that we eventually want to update.

Related:
https://github.com/tlaplus/tlaplus/pull/1088#discussion_r1876749598

[Test][TLC]

### CODE REVIEW

**Comment by lemmy** [comment]
Enabled by:
* https://github.com/tlaplus/CommunityModules/commit/e88d04f0d8caa2c5ce97849a6d33ce08c969baff
* https://github.com/tlaplus/CommunityModules/commit/e43de9c902a1d84744509592d9380c4bd9335751

**Comment by ahelwer** [comment]
The tests are pulled into the final release jar, aren't they?

**Comment by lemmy** [comment]
The CM jar is not included in the final release. However, there are tests and functionalities that will fail if the CM jar is absent when a user executes the tests or functionalities using `tla2tools.jar`.

File: tlatools/org.lamport.tlatools/lib/CommunityModules.jar
Comment (Calvin-L) [inline tlatools/org.lamport.tlatools/lib/CommunityModules.jar:1]: We should add a note about this jar's origin:
 - What version of the Community Modules is this?
 - Does this jar include the dependencies (i.e. is it the `CommunityModules.jar` or `CommunityModules-deps.jar`)?

We could either pack that information into the filename (by calling it e.g. `CommunityModules-2024XYZ-nodeps.jar`) or in a README somewhere.

Comment (lemmy) [reply tlatools/org.lamport.tlatools/lib/CommunityModules.jar:1]: https://github.com/tlaplus/tlaplus/pull/1103/commits/fd0044d5afb05f81bb6dceb26a9a7263e710cdaf amends the git commit message and adds a note.

**Comment by lemmy** [comment]
Merging this since it addresses @fhackett (https://github.com/tlaplus/tlaplus/pull/1102#issuecomment-2556402206) and my  (#1088) use case.

**Pull request merged**